### PR TITLE
[NU-1291] [Security] Processing type data - check read access to category

### PR DESCRIPTION
**Pull request by arkadius**
## Checklist before merge
- [x] Related issue ID is placed at the beginning of PR title in \[brackets\] (can be GH issue or Nu Jira issue)
- [x] Code is cleaned from temporary changes and commented out lines
- [x] Parts of the code that are not easy to understand are documented in the code
- [x] Changes are covered by automated tests
- [ ] Showcase in dev-application.conf added to demonstrate the feature
- [x] Documentation added or updated
- [x] Added entry in _Changelog.md_ describing the change from the perspective of a public distribution user
- [x] Added _MigrationGuide.md_ entry in the appropriate subcategory if introducing a breaking change
- [ ] Verify that PR will be squashed during merge

### CODE REVIEW

File: designer/server/src/main/scala/pl/touk/nussknacker/ui/services/AppApiHttpService.scala [hunk 1]
```diff
@@ -97,8 +97,10 @@ class AppApiHttpService(
         Future {
           import net.ceedubs.ficus.Ficus._
           val configuredBuildInfo = config.getAs[Map[String, String]]("globalBuildInfo")
+          // TODO: Warning, here is a little security leak. Everyone can discover configured processing types.
+          //       We should consider authorization of access to this data
```
Comment (pielas) [inline designer/server/src/main/scala/pl/touk/nussknacker/ui/services/AppApiHttpService.scala:101]:
```suggestion
          //       We should consider adding an authorization of access rights to this data.
```

File: designer/server/src/main/scala/pl/touk/nussknacker/ui/statistics/UsageStatisticsReportsSettingsDeterminer.scala [hunk 2]
```diff
@@ -69,13 +70,16 @@ class UsageStatisticsReportsSettingsDeterminer(
   }
 
   private[statistics] def determineQueryParams(): ListMap[String, String] = {
-    val deploymentManagerTypes = processingTypeStatistics.values.map(_.deploymentManagerType).map {
+    // TODO: Warning, here is a security leak. We report statistics in the scope of processing types to which
+    //       given user has no access.
```
Comment (pielas) [inline designer/server/src/main/scala/pl/touk/nussknacker/ui/statistics/UsageStatisticsReportsSettingsDeterminer.scala:74]:
```suggestion
    //       given user has no access rights.
```

**Review by pielas — CHANGES_REQUESTED** [review]
I think it would be generally

File: designer/server/src/main/scala/pl/touk/nussknacker/ui/process/processingtypedata/ProcessingTypeDataProvider.scala [hunk 3]
```diff
@@ -20,23 +27,27 @@ import pl.touk.nussknacker.engine.api.process.ProcessingType
...
 
   // TODO: replace with proper forType handling
-  def forTypeUnsafe(typ: ProcessingType): T = forType(typ)
+  final def forTypeUnsafe(typ: ProcessingType)(implicit user: LoggedUser): T = forType(typ)
     .getOrElse(throw new IllegalArgumentException(s"Unknown typ: $typ, known types are: ${all.keys.mkString(", ")}"))
 
-  def all: Map[ProcessingType, T]
+  def all(implicit user: LoggedUser): Map[ProcessingType, T]
 
+  // TODO: We should return type that can produce views for users with access to certain categories. Thanks to that
```
[the remark this answers is not in the document]
Comment (arkadius) [reply designer/server/src/main/scala/pl/touk/nussknacker/ui/process/processingtypedata/ProcessingTypeDataProvider.scala:38]: I haven't in mind `ProcessingType` here. It was about generic parameter `C` (for combined type). I'll change it a bit to make it more clear.

**Review by pielas — APPROVED** [review]
LGTM

**Pull request merged**